### PR TITLE
MISRA-C fixes

### LINKED ISSUE #9895: MISRA C Add 'u' or 'U' suffix for all unsigned integer constants
All unsigned constants must have either a "u" or "U" suffix. Generated code (e.g generated_dts_board.h) is not adding it.

part of:  #9552

### PR DESCRIPTION
**Pull request by ceolin**
#9895
#9886 
#9889

### CODE REVIEW

File: include/logging/log_core.h [hunk 1]
```diff
@@ -219,13 +219,13 @@ extern "C" {
 				.source_id = _id			    \
 			};						    \
 									    \
-			if ((1 << _level) & LOG_FUNCTION_PREFIX_MASK) {	    \
+			if (BIT(_level) & LOG_FUNCTION_PREFIX_MASK) {	\
```
Comment (andrewboie) [inline include/logging/log_core.h]: should this line also have != 0 instead of implicit boolean conversion?

**Review by tyrantlizard77 — APPROVED** [review]

File: include/logging/log_core.h (hunk 1 above)
Comment (ceolin) [reply include/logging/log_core.h]: yes, thanks for spotting it, will fix

**Comment by nashif** [comment]
> #9895
> #9886
> #9889

what about those? Does it fix them? fully or partially? Please add more context and use "Fixes #<number>" to enable auto-losing when this is merged.

**Comment by ceolin** [comment]
> > #9895
> > #9886
> > #9889
> 
> what about those? Does it fix them? fully or partially? Please add more context and use "Fixes #" to enable auto-losing when this is merged.

They partially fix them. There are still violations, some of them will probably be addressed in deviations.
I'm adding the MISRA-C rule in commit message to make easier identify commits fixing MISRA-C.

Would like me to add "FIXES" tag in commit message even not fully fixing the issue ?

**Review by galak — APPROVED** [review]

File: include/posix/pthread.h [hunk 2]
```diff
@@ -54,8 +54,8 @@ struct posix_thread {
 
 /* Pthread cancellation */
 #define _PTHREAD_CANCEL_POS	0
-#define PTHREAD_CANCEL_ENABLE	(0 << _PTHREAD_CANCEL_POS)
-#define PTHREAD_CANCEL_DISABLE	(1 << _PTHREAD_CANCEL_POS)
+#define PTHREAD_CANCEL_ENABLE	BIT(_PTHREAD_CANCEL_POS)
+#define PTHREAD_CANCEL_DISABLE	BIT(_PTHREAD_CANCEL_POS)
```
Comment (MaureenHelm) [inline include/posix/pthread.h]: This is changing the value of `PTHREAD_CANCEL_ENABLE`

**Review by galak — DISMISSED** [review]
Please rebase, and address the issue around PTHREAD_CANCEL_ENABLE

File: include/posix/pthread.h (hunk 2 above)
Comment (ceolin) [reply include/posix/pthread.h]: @MaureenHelm indeed, thanks for pointing it out.

**Comment by ceolin** [comment]
@galak done :)

**Comment by ceolin** [comment]
@galak @nashif I got a timeout in our CI, can one of you trigger a new build to see if it happens again ?

**Review by nashif — APPROVED** [review]

**Pull request merged**